front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -42,7 +48,22 @@ const ManchetteWithSpaceTimeChartWrapper = ({
   );
 
   const [showSettingsPanel, setShowSettingsPanel] = useState(false);
-  const [settings, setSettings] = useState({ showConflicts: false });
+  const [settings, setSettings] = useState({
+    showConflicts: false,
+    showSignalsStates: false,
+  });
+
+  const occupancyBlocks = projectPathTrainResult.flatMap((train) => {
+    const departureTime = new Date(train.departure_time).getTime();
+
+    return train.signal_updates.map((block) => ({
+      timeStart: departureTime + block.time_start,
+      timeEnd: departureTime + block.time_end,
+      spaceStart: block.position_start,
+      spaceEnd: block.position_end,
+      color: ASPECT_LABELS_COLORS[block.aspect_label as AspectLabel] || '',
```

Comment:
Nit: I'd slightly prefer to assert that `ASPECT_LABELS_COLORS` contains a valid color with `!` instead of `|| ''` because it would be a frontend bug to miss one of these.